mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -407,6 +449,103 @@ private[classification] class MultiClassSummarizer extends Serializable {
   }
 }
 
+@Experimental
+/**
+ * :: Experimental ::
+ * Logistic regression training results.
+ * @param predictions dataframe outputted by the model's `transform` method.
+ * @param probabilityCol field in "predictions" which gives the calibrated probability of
+ *                       each sample as a vector.
+ * @param labelCol field in "predictions" which gives the true label of each sample.
+ * @param objectiveHistory objective function (scaled loss + regularization) at each iteration.
+ */
+class LogisticRegressionTrainingSummary private[classification] (
+    predictions: DataFrame,
+    probabilityCol: String,
+    labelCol: String,
+    val objectiveHistory: Array[Double])
+  extends LogisticRegressionSummary(predictions, probabilityCol, labelCol) {
+
+  /** Number of training iterations until termination */
+  val totalIterations = objectiveHistory.length
+
+}
+
+@Experimental
+/**
+ * :: Experimental ::
+ * Logistic regression results for a given model.
+ * @param predictions dataframe outputted by the model's `transform` method.
+ * @param probabilityCol field in "predictions" which gives the calibrated probability of
+ *                       each sample.
+ * @param labelCol field in "predictions" which gives the true label of each sample.
+ */
+class LogisticRegressionSummary private[classification] (
```

Comment:
Before merging, I still think we need to create the abstraction to handle both binary and multiclass classification summaries.  If we commit this PR now, then we will need to break the API in the future to support multiclass classification.  Instead, we should prepare for it now by providing the abstraction + the binary summary in this PR.  (It's fine not to have the multiclass version in this PR.)